process/smartContract/process.go

```diff
@@ -403,6 +403,19 @@ func (sc *scProcessor) processVMOutput(
 		return nil, err
 	}
 
+	//we need to reload the sender account as in the case of refunding the exact account that was previously
+	//modified in saveSCOutputToCurrentState, the modifications done there should be visible here
+	//this should be applied only on sender accounts from current shard
+	if acntSnd != nil && !acntSnd.IsInterfaceNil() {
```

Comment:
Maybe refactor a bit since we have 3 levels here?

Reply:
Done, changed to a function. Code should be more readable.